index.js

```diff
@@ -47,41 +53,95 @@ function wrapStaticMethods(klass, methodNames) {
+  wrapFunctions(klass.prototype, options, false);
+}
+
+/**
+ * Helper function that conforms input arguments to a single options object.
+ *
+ * @param   {string[]} [methodNames] Optional array of method names
+ * @param   {Object}   [options] Optional options
+ * @returns {Object}   options Options object
+ * @throws  {Error}    If methodNames is provided, but is not an array,
+ *                     or options is provided, but is not an object,
+ *                     or wrapper is provided, but is not a function,
+ *                     or asyncWrapper is provided, but is not a function
+ */
+function conformOptions(methodNames, options) {
+  if (!options && methodNames && !(methodNames instanceof Array)) {
+    options = methodNames;
+    methodNames = undefined;
+  }
+
+  if (!options) options = {};
+  if (typeof options != 'object' || options instanceof Array || options instanceof Date) throw new Error('Optional options should be an object if provided');
+  options = cloneOptions(options);
+
+  if (methodNames) options.methodNames = methodNames;
+  if (options.methodNames && !(options.methodNames instanceof Array)) throw new Error('Optional methodNames should be an array if provided');
+
+  if (options.wrapper === undefined) options.wrapper = Promise.coroutine;
+  if (options.wrapper && typeof options.wrapper != 'function') throw new Error('Optional wrapper should be a function if provided');
+
+  if (options.asyncWrapper === undefined) options.asyncWrapper = Promise.method;
+  if (options.asyncWrapper && typeof options.asyncWrapper != 'function') throw new Error('Optional asyncWrapper should be a function if provided');
+
+  if (options.asyncWrapCondition === undefined) options.asyncWrapCondition = asyncWrapCondition;
+  if (options.asyncWrapCondition && typeof options.asyncWrapCondition != 'function') throw new Error('Optional asyncWrapCondition should be a function if provided');
+
+  return options;
+}
+
 /**
- * Helper function that validates the methodNames parameter.
+ * Helper function that clones options object.
  *
- * @param {string[]} [methodNames] Optional array of method names
- * @throws {Error}   If methodNames is provided, but is not an array
+ * @param   {Object} options Options object
+ * @returns {Object} Shallow clone of options object
  */
-function validateMethodNames(methodNames) {
-  if (methodNames && !(methodNames instanceof Array)) {
-    throw new Error('Optional methodNames should be an array if provided');
+function cloneOptions(options) {
+  var out = {};
+  for (var key in options) {
+    out[key] = options[key];
   }
+  return out;
+}
+
+/**
+ * Default asyncWrapCondition function.
+ * Returns `true` if method should be wrapped, `false` if not.
+ *
+ * @param {string} key Method name
+ * @returns {boolean}
+ */
+function asyncWrapCondition(key) {
+    return key.endsWith('Async');
 }
 
-function wrapFunctions(target, methodNames) {
+function wrapFunctions(target, options, isStatic) {
```

Comment:
Looks like you switched to using the leading-underscore naming convention for pricate helpers, so this should probably be `_wrapFunctions` ?